src/main/java/ai/elimu/logic/LicenseGenerator.java

```diff
@@ -0,0 +1,43 @@
+package ai.elimu.logic;
+
+import ai.elimu.model.project.AppCollection;
+import ai.elimu.model.project.Project;
+
+/**
+ * Utility class for generating license numbers to be used for unlocking access to {@link AppCollection}s 
+ * in custom {@link Project}s
+ */
+public class LicenseGenerator {
+    
+    private static final int LICENSE_LENGTH = 16;
+    
+    private static final String GROUP_SEPARATOR = "-";
+    
+    /**
+     * Ambiguous characters '0' an '1' are not included.
+     */
+    private static final String VALID_CHARACTERS = "abcdef23456789";
+    
+    /**
+     * Generate license number on the format "bddf-d8f4-2adf-a365".
```

Comment:
@tuancoltech Note that a license number on this format will be typed into the Appstore application, if pressing "Yes" on the first screen.

![device-2017-11-11-103155](https://user-images.githubusercontent.com/1451036/32688133-93a099f2-c6cb-11e7-8581-00360d896397.png)